# Eerste verkenning en validatie van de observation details. 
Verkennen van de data en uitvoeren van een eerste ruwe cleaning. </br>
Het resultaat wordt geplaatst in de sectie 'cleaned'

In [19]:
import pandas as pd
import os

In [20]:
species_id =70
folder_path_clean = "./scraped_data/cleaned/"

In [21]:
os.getcwd()
# set the max columns to none
pd.set_option('display.max_columns', None)
# set the max columns to none
pd.set_option('display.max_rows', None)


## Verkenning van de data


In [22]:
# file path
observation_details = f'scraped_data/observation_details_{species_id}_complete.csv'


# Load the data
df = pd.read_csv(observation_details, skipinitialspace=True, index_col=0)
df.head()

species_id species_name species_name_scientific  \
observation_id                                                    
336194665             70.0   Boomklever          Sitta europaea   
336207548             70.0   Boomklever          Sitta europaea   
336207346             70.0   Boomklever          Sitta europaea   
336199027             70.0   Boomklever          Sitta europaea   
336207352             70.0   Boomklever          Sitta europaea   

                                          validation  gps_coordinates  \
observation_id                                                          
336194665       Goedgekeurd (automatische validatie)  51.1143, 4.2675   
336207548       Goedgekeurd (automatische validatie)  51.0751, 5.0718   
336207346       Goedgekeurd (automatische validatie)  50.5213, 5.3332   
336199027       Goedgekeurd (automatische validatie)  51.0177, 5.1856   
336207352       Goedgekeurd (automatische validatie)  50.5162, 5.3350   

               accuracy   source              date amount         life_stage  \
observation_id                                                                 
336194665           15m     Site  2024-12-31 15:10      2  adult winterkleed   
336207548           15m     Site  2024-12-31 15:00      1           onbekend   
336207346            5m  ObsMapp  2024-12-31 14:26      1           onbekend   
336199027           18m     iObs        2024-12-31      1           onbekend   
336207352            5m  ObsMapp  2024-12-31 13:56      1           onbekend   

                   activity  location_id                        location  \
observation_id                                                             
336194665       foeragerend      29110.0  Hingene - Schellandpolder (AN)   
336207548       ter plaatse     675306.0     Tessenderlo - Baalberg (LI)   
336207346           roepend      42124.0                Outrelouxhe (LG)   
336199027       ter plaatse      28673.0      Lummen - Laren - Zuid (LI)   
336207352       foeragerend      42124.0                Outrelouxhe (LG)   

                observer_id                 observer_name counting_method  \
observation_id                                                              
336194665           99836.0           Van Osselaer Philip            geen   
336207548          149793.0                  Elly Gombeer            geen   
336207346           43403.0  Jean-Sébastien Rousseau-Piot        onbekend   
336199027          118615.0                R. Vansweevelt        onbekend   
336207352           43403.0  Jean-Sébastien Rousseau-Piot        onbekend   

                           method  
observation_id                     
336194665       gezien en gehoord  
336207548                  gezien  
336207346                 gehoord  
336199027                onbekend  
336207352                onbekend

In [23]:
df.index 

Index([336194665, 336207548, 336207346, 336199027, 336207352, 336193003,
       336188047, 336193038, 336193048, 336197798,
       ...
        45654126,  45654069,  68459523,  41660038,  41659963,  74938220,
        62234945,  96686569,  43875950,  47673787],
      dtype='int64', name='observation_id', length=277033)

In [24]:
df.columns

Index(['species_id', 'species_name', 'species_name_scientific', 'validation',
       'gps_coordinates', 'accuracy', 'source', 'date', 'amount', 'life_stage',
       'activity', 'location_id', 'location', 'observer_id', 'observer_name',
       'counting_method', 'method'],
      dtype='object')

In [25]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 277033 entries, 336194665 to 47673787
Data columns (total 17 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   species_id               277029 non-null  float64
 1   species_name             277029 non-null  object 
 2   species_name_scientific  277029 non-null  object 
 3   validation               277029 non-null  object 
 4   gps_coordinates          269863 non-null  object 
 5   accuracy                 269757 non-null  object 
 6   source                   259132 non-null  object 
 7   date                     277029 non-null  object 
 8   amount                   277029 non-null  object 
 9   life_stage               276552 non-null  object 
 10  activity                 276552 non-null  object 
 11  location_id              272738 non-null  float64
 12  location                 272738 non-null  object 
 13  observer_id              276171 non-null  float64
 14 

In [26]:
stats = df.describe(include='all')
stats

species_id species_name species_name_scientific  \
count   277029.000000       277029                  277029   
unique            NaN            3                       3   
top               NaN   Boomklever          Sitta europaea   
freq              NaN       276537                  276537   
mean       529.815568          NaN                     NaN   
std      10901.305917          NaN                     NaN   
min         70.000000          NaN                     NaN   
25%         70.000000          NaN                     NaN   
50%         70.000000          NaN                     NaN   
75%         70.000000          NaN                     NaN   
max     258978.000000          NaN                     NaN   

                                  validation  gps_coordinates accuracy  \
count                                 277029           269863   269757   
unique                                     5           216560      656   
top     Goedgekeurd (automatische validatie)  50.9686, 5.3391      25m   
freq                                  242999              711    69431   
mean                                     NaN              NaN      NaN   
std                                      NaN              NaN      NaN   
min                                      NaN              NaN      NaN   
25%                                      NaN              NaN      NaN   
50%                                      NaN              NaN      NaN   
75%                                      NaN              NaN      NaN   
max                                      NaN              NaN      NaN   

         source        date  amount life_stage     activity    location_id  \
count    259132      277029  277029     276552       276552  272738.000000   
unique       27      172237     248         20           48            NaN   
top     ObsMapp  2013-04-07       1   onbekend  ter plaatse            NaN   
freq     111400          94  232976     253834       146999            NaN   
mean        NaN         NaN     NaN        NaN          NaN   83370.406775   
std         NaN         NaN     NaN        NaN          NaN  128462.203471   
min         NaN         NaN     NaN        NaN          NaN   23089.000000   
25%         NaN         NaN     NaN        NaN          NaN   28734.000000   
50%         NaN         NaN     NaN        NaN          NaN   31413.000000   
75%         NaN         NaN     NaN        NaN          NaN   72759.000000   
max         NaN         NaN     NaN        NaN          NaN  724090.000000   

                                                 location   observer_id  \
count                                              272738  2.761710e+05   
unique                                               8062           NaN   
top     Eeklo - Het Leen - Noord (4250A) (Prov. Dom. O...           NaN   
freq                                                 4079           NaN   
mean                                                  NaN  8.393163e+04   
std                                                   NaN  9.219410e+04   
min                                                   NaN  1.140000e+02   
25%                                                   NaN  4.088300e+04   
50%                                                   NaN  4.972200e+04   
75%                                                   NaN  9.728500e+04   
max                                                   NaN  1.060523e+06   

       observer_name counting_method    method  
count         276171          277029    277029  
unique          9728               7        18  
top       D. Peeters        onbekend  onbekend  
freq            9265          242533    160932  
mean             NaN             NaN       NaN  
std              NaN             NaN       NaN  
min              NaN             NaN       NaN  
25%              NaN             NaN       NaN  
50%              NaN             NaN       NaN  
75%              NaN             NaN       NaN  


In [27]:
df.isnull().sum()

species_id                     4
species_name                   4
species_name_scientific        4
validation                     4
gps_coordinates             7170
accuracy                    7276
source                     17901
date                           4
amount                         4
life_stage                   481
activity                     481
location_id                 4295
location                    4295
observer_id                  862
observer_name                862
counting_method                4
method                         4
dtype: int64

# First cleaning

## Ongeldige species_id (NaN)

In [28]:
df[df['species_id'].isnull()]

species_id species_name species_name_scientific validation  \
observation_id                                                               
211495505              NaN          NaN                     NaN        NaN   
210627812              NaN          NaN                     NaN        NaN   
210627870              NaN          NaN                     NaN        NaN   
203114064              NaN          NaN                     NaN        NaN   

               gps_coordinates accuracy source date amount life_stage  \
observation_id                                                          
211495505                  NaN      NaN    NaN  NaN    NaN        NaN   
210627812                  NaN      NaN    NaN  NaN    NaN        NaN   
210627870                  NaN      NaN    NaN  NaN    NaN        NaN   
203114064                  NaN      NaN    NaN  NaN    NaN        NaN   

               activity  location_id location  observer_id observer_name  \
observation_id                                                             
211495505           NaN          NaN      NaN          NaN           NaN   
210627812           NaN          NaN      NaN          NaN           NaN   
210627870           NaN          NaN      NaN          NaN           NaN   
203114064           NaN          NaN      NaN          NaN           NaN   

               counting_method method  
observation_id                         
211495505                  NaN    NaN  
210627812                  NaN    NaN  
210627870                  NaN    NaN  
203114064                  NaN    NaN

De waarnemingen (observation_id) met species_id Nan  

Halsbandparkiet (species_id = 116)  
- 322156271  

Boomklever (species_id = 70)  
- 211495505  
- 210627812  
- 210627870  
- 203114064  

Als we deze verifiëren op de website eg. (https://waarnemingen.be/observation/322156271/), dan zien we dat deze inderdaad niet bestaan.  
Een verklaring hiervoor is dat deze nog wel bestonden toen we de general data ophaalden, maar nadien verwijderd zijn geweest. Vermits deze waarnemingen niet meer bestaan, verwijderen we deze ook uit onze dataset.

In [ ]:
df_first_cleaning = df.dropna(subset=['species_id'])

species_id species_name species_name_scientific  \
count   277029.000000       277029                  277029   
unique            NaN            3                       3   
top               NaN   Boomklever          Sitta europaea   
freq              NaN       276537                  276537   
mean       529.815568          NaN                     NaN   
std      10901.305917          NaN                     NaN   
min         70.000000          NaN                     NaN   
25%         70.000000          NaN                     NaN   
50%         70.000000          NaN                     NaN   
75%         70.000000          NaN                     NaN   
max     258978.000000          NaN                     NaN   

                                  validation  gps_coordinates accuracy  \
count                                 277029           269863   269757   
unique                                     5           216560      656   
top     Goedgekeurd (automatische validatie)  50.9686, 5.3391      25m   
freq                                  242999              711    69431   
mean                                     NaN              NaN      NaN   
std                                      NaN              NaN      NaN   
min                                      NaN              NaN      NaN   
25%                                      NaN              NaN      NaN   
50%                                      NaN              NaN      NaN   
75%                                      NaN              NaN      NaN   
max                                      NaN              NaN      NaN   

         source        date  amount life_stage     activity    location_id  \
count    259132      277029  277029     276552       276552  272738.000000   
unique       27      172237     248         20           48            NaN   
top     ObsMapp  2013-04-07       1   onbekend  ter plaatse            NaN   
freq     111400          94  232976     253834       146999            NaN   
mean        NaN         NaN     NaN        NaN          NaN   83370.406775   
std         NaN         NaN     NaN        NaN          NaN  128462.203471   
min         NaN         NaN     NaN        NaN          NaN   23089.000000   
25%         NaN         NaN     NaN        NaN          NaN   28734.000000   
50%         NaN         NaN     NaN        NaN          NaN   31413.000000   
75%         NaN         NaN     NaN        NaN          NaN   72759.000000   
max         NaN         NaN     NaN        NaN          NaN  724090.000000   

                                                 location   observer_id  \
count                                              272738  2.761710e+05   
unique                                               8062           NaN   
top     Eeklo - Het Leen - Noord (4250A) (Prov. Dom. O...           NaN   
freq                                                 4079           NaN   
mean                                                  NaN  8.393163e+04   
std                                                   NaN  9.219410e+04   
min                                                   NaN  1.140000e+02   
25%                                                   NaN  4.088300e+04   
50%                                                   NaN  4.972200e+04   
75%                                                   NaN  9.728500e+04   
max                                                   NaN  1.060523e+06   

       observer_name counting_method    method  
count         276171          277029    277029  
unique          9728               7        18  
top       D. Peeters        onbekend  onbekend  
freq            9265          242533    160932  
mean             NaN             NaN       NaN  
std              NaN             NaN       NaN  
min              NaN             NaN       NaN  
25%              NaN             NaN       NaN  
50%              NaN             NaN       NaN  
75%              NaN             NaN       NaN  


## Species_id
De species_id is niet degene die we verwachten. Bij de general observations hebben we op een specifieke soort gezocht, maar nu blijkt dat bij de observation_details de boomklever, dat deze niet altijd overeenkomt met degene die we wilden ophalen. </br>

#### conclusie
Bij de boomklevers zien we dat we ook een species_id 258977 en 258978 hebben. Dit blijken ondersoorten van de boomklever te zijn , nl Boomklever ssp caesia en Boomklever ssp europaea </br>
In onze oefening gaan we het onderscheid tussen deze soorten niet maken, deze ondersoort blijft behouden in de data.

In [42]:
df_first_cleaning.groupby('species_id').nunique()

species_name  species_name_scientific  validation  \
species_id                                                      
70.0                   1                        1           5   
258977.0               1                        1           3   
258978.0               1                        1           2   

            gps_coordinates  accuracy  source    date  amount  life_stage  \
species_id                                                                  
70.0                 216213       656      27  171871     248          20   
258977.0                397        31       7     480      11           5   
258978.0                  3         3       2       3       1           1   

            activity  location_id  location  observer_id  observer_name  \
species_id                                                                
70.0              48         8072      8055         9874           9688   
258977.0          13          282       282          152            152   
258978.0           3            3         3            3              3   

            counting_method  method  
species_id                           
70.0                      7      18  
258977.0                  6       5  
258978.0                  2       1

In [41]:
df1 = df_first_cleaning[df_first_cleaning['species_id'] == 70].head(2)
df2 = df_first_cleaning[df_first_cleaning['species_id'] == 258977.0].head(2)
df3 = df_first_cleaning[df_first_cleaning['species_id'] == 258978.0].head(2)

pd.concat([df1, df2, df3], axis=0)

species_id             species_name  species_name_scientific  \
observation_id                                                                 
336194665             70.0               Boomklever           Sitta europaea   
336207548             70.0               Boomklever           Sitta europaea   
335598923         258977.0    Boomklever ssp caesia    Sitta europaea caesia   
335598990         258977.0    Boomklever ssp caesia    Sitta europaea caesia   
323674036         258978.0  Boomklever ssp europaea  Sitta europaea europaea   
309150269         258978.0  Boomklever ssp europaea  Sitta europaea europaea   

                                          validation  gps_coordinates  \
observation_id                                                          
336194665       Goedgekeurd (automatische validatie)  51.1143, 4.2675   
336207548       Goedgekeurd (automatische validatie)  51.0751, 5.0718   
335598923                                   Onbekend  50.3485, 5.4071   
335598990                                   Onbekend  50.3478, 5.4100   
323674036                   Goedgekeurd (met bewijs)  50.8674, 4.6781   
309150269                   Goedgekeurd (met bewijs)  51.1074, 5.2000   

               accuracy       source              date amount  \
observation_id                                                  
336194665           15m         Site  2024-12-31 15:10      2   
336207548           15m         Site  2024-12-31 15:00      1   
335598923            4m      ObsMapp  2024-12-21 10:01      1   
335598990            3m      ObsMapp  2024-12-20 10:27      1   
323674036            7m  ObsIdentify  2024-08-11 10:49      1   
309150269           15m  ObsIdentify  2024-05-12 12:41      1   

                       life_stage       activity  location_id  \
observation_id                                                  
336194665       adult winterkleed    foeragerend      29110.0   
336207548                onbekend    ter plaatse     675306.0   
335598923                onbekend        roepend      42334.0   
335598990                onbekend    foeragerend      42334.0   
323674036                onbekend  vondst (dood)      30919.0   
309150269                onbekend    ter plaatse      27563.0   

                                                 location  observer_id  \
observation_id                                                           
336194665                  Hingene - Schellandpolder (AN)      99836.0   
336207548                     Tessenderlo - Baalberg (LI)     149793.0   
335598923                                     Septon (LX)      42518.0   
335598990                                     Septon (LX)      42518.0   
323674036       Heverlee - Egenhoven - Egenhoven Bos (VB)     854160.0   
309150269                                    Oostham (LI)     840139.0   

                      observer_name counting_method             method  
observation_id                                                          
336194665       Van Osselaer Philip            geen  gezien en gehoord  
336207548              Elly Gombeer            geen             gezien  
335598923           Jorg Lambrechts    exact aantal  gezien en gehoord  
335598990           Jorg Lambrechts    exact aantal             gezien  
323674036                      Nora            geen           onbekend  
309150269                  Kristian            geen           onbekend

In [31]:
stats = df_first_cleaning.describe(include='all')
stats

species_id species_name species_name_scientific  \
count   277029.000000       277029                  277029   
unique            NaN            3                       3   
top               NaN   Boomklever          Sitta europaea   
freq              NaN       276537                  276537   
mean       529.815568          NaN                     NaN   
std      10901.305917          NaN                     NaN   
min         70.000000          NaN                     NaN   
25%         70.000000          NaN                     NaN   
50%         70.000000          NaN                     NaN   
75%         70.000000          NaN                     NaN   
max     258978.000000          NaN                     NaN   

                                  validation  gps_coordinates accuracy  \
count                                 277029           269863   269757   
unique                                     5           216560      656   
top     Goedgekeurd (automatische validatie)  50.9686, 5.3391      25m   
freq                                  242999              711    69431   
mean                                     NaN              NaN      NaN   
std                                      NaN              NaN      NaN   
min                                      NaN              NaN      NaN   
25%                                      NaN              NaN      NaN   
50%                                      NaN              NaN      NaN   
75%                                      NaN              NaN      NaN   
max                                      NaN              NaN      NaN   

         source        date  amount life_stage     activity    location_id  \
count    259132      277029  277029     276552       276552  272738.000000   
unique       27      172237     248         20           48            NaN   
top     ObsMapp  2013-04-07       1   onbekend  ter plaatse            NaN   
freq     111400          94  232976     253834       146999            NaN   
mean        NaN         NaN     NaN        NaN          NaN   83370.406775   
std         NaN         NaN     NaN        NaN          NaN  128462.203471   
min         NaN         NaN     NaN        NaN          NaN   23089.000000   
25%         NaN         NaN     NaN        NaN          NaN   28734.000000   
50%         NaN         NaN     NaN        NaN          NaN   31413.000000   
75%         NaN         NaN     NaN        NaN          NaN   72759.000000   
max         NaN         NaN     NaN        NaN          NaN  724090.000000   

                                                 location   observer_id  \
count                                              272738  2.761710e+05   
unique                                               8062           NaN   
top     Eeklo - Het Leen - Noord (4250A) (Prov. Dom. O...           NaN   
freq                                                 4079           NaN   
mean                                                  NaN  8.393163e+04   
std                                                   NaN  9.219410e+04   
min                                                   NaN  1.140000e+02   
25%                                                   NaN  4.088300e+04   
50%                                                   NaN  4.972200e+04   
75%                                                   NaN  9.728500e+04   
max                                                   NaN  1.060523e+06   

       observer_name counting_method    method  
count         276171          277029    277029  
unique          9728               7        18  
top       D. Peeters        onbekend  onbekend  
freq            9265          242533    160932  
mean             NaN             NaN       NaN  
std              NaN             NaN       NaN  
min              NaN             NaN       NaN  
25%              NaN             NaN       NaN  
50%              NaN             NaN       NaN  
75%              NaN             NaN       NaN  


# Gebruikte waarden en hun frequentie checken

#### conclusie
- amount zal nog moeten worden opgekuist naderhand, want bevat niet enkel het aantal vogels, maar ook tekst
- de observatiedatums starten al van in 1930, we zullen moeten beslissen of we deze willen opnemen in onze dataset



In [32]:
# Kijken welke mogelijke waarden er zijn, en wat de frequentie in de andere kolommen is
df_first_cleaning.groupby('validation').nunique()

species_id  species_name  \
validation                                                       
(nog) niet te beoordelen                       2             2   
Goedgekeurd (aannemelijk)                      2             2   
Goedgekeurd (automatische validatie)           1             1   
Goedgekeurd (met bewijs)                       3             3   
Onbekend                                       2             2   

                                      species_name_scientific  \
validation                                                      
(nog) niet te beoordelen                                    2   
Goedgekeurd (aannemelijk)                                   2   
Goedgekeurd (automatische validatie)                        1   
Goedgekeurd (met bewijs)                                    3   
Onbekend                                                    2   

                                      gps_coordinates  accuracy  source  \
validation                                                                
(nog) niet te beoordelen                          111        17       4   
Goedgekeurd (aannemelijk)                       15415       189      18   
Goedgekeurd (automatische validatie)           193507       569      26   
Goedgekeurd (met bewijs)                         5908       164      14   
Onbekend                                         4896       156      19   

                                        date  amount  life_stage  activity  \
validation                                                                   
(nog) niet te beoordelen                 150      11           3        15   
Goedgekeurd (aannemelijk)              10257      46          12        37   
Goedgekeurd (automatische validatie)  156014     201          20        35   
Goedgekeurd (met bewijs)                6084      42          13        31   
Onbekend                                6964      42           9        33   

                                      location_id  location  observer_id  \
validation                                                                 
(nog) niet te beoordelen                       97        97           74   
Goedgekeurd (aannemelijk)                    2456      2455          890   
Goedgekeurd (automatische validatie)         7511      7495         8904   
Goedgekeurd (met bewijs)                     1965      1964         2245   
Onbekend                                     2121      2120         1376   

                                      observer_name  counting_method  method  
validation                                                                    
(nog) niet te beoordelen                         74                5       7  
Goedgekeurd (aannemelijk)                       886                6      10  
Goedgekeurd (automatische validatie)           8776                7      18  
Goedgekeurd (met bewijs)                       2231                5      10  
Onbekend                                       1374                7      10

In [43]:
df_first_cleaning.groupby('species_name').nunique()

species_id  species_name_scientific  validation  \
species_name                                                               
Boomklever                        1                        1           5   
Boomklever ssp caesia             1                        1           3   
Boomklever ssp europaea           1                        1           2   

                         gps_coordinates  accuracy  source    date  amount  \
species_name                                                                 
Boomklever                        216213       656      27  171871     248   
Boomklever ssp caesia                397        31       7     480      11   
Boomklever ssp europaea                3         3       2       3       1   

                         life_stage  activity  location_id  location  \
species_name                                                           
Boomklever                       20        48         8072      8055   
Boomklever ssp caesia             5        13          282       282   
Boomklever ssp europaea           1         3            3         3   

                         observer_id  observer_name  counting_method  method  
species_name                                                                  
Boomklever                      9874           9688                7      18  
Boomklever ssp caesia            152            152                6       5  
Boomklever ssp europaea            3              3                2       1

In [ ]:
df_first_cleaning.groupby('amount').nunique().head(10)

species_id  species_name  \
amount                                                                         
1                                                            3             3   
1\n            \n            \n\n\n\n\n\n    \n...           1             1   
1\n            \n            \n\n\n\n\n\n    \n...           1             1   
1\n            \n            \n\n\n\n\n\n    \n...           1             1   
1\n            \n            \n\n\n\n\n\n    \n...           1             1   
1\n            \n            \n\n\n\n\n\n    \n...           1             1   
1\n            \n            \n\n\n\n\n\n    \n...           1             1   
1\n            \n            \n\n\n\n\n\n    \n...           1             1   
1\n            \n            \n\n\n\n\n\n    \n...           1             1   
1\n            \n            \n\n\n\n\n\n    \n...           1             1   

                                                    species_name_scientific  \
amount                                                                        
1                                                                         3   
1\n            \n            \n\n\n\n\n\n    \n...                        1   
1\n            \n            \n\n\n\n\n\n    \n...                        1   
1\n            \n            \n\n\n\n\n\n    \n...                        1   
1\n            \n            \n\n\n\n\n\n    \n...                        1   
1\n            \n            \n\n\n\n\n\n    \n...                        1   
1\n            \n            \n\n\n\n\n\n    \n...                        1   
1\n            \n            \n\n\n\n\n\n    \n...                        1   
1\n            \n            \n\n\n\n\n\n    \n...                        1   
1\n            \n            \n\n\n\n\n\n    \n...                        1   

                                                    validation  \
amount                                                           
1                                                            5   
1\n            \n            \n\n\n\n\n\n    \n...           1   
1\n            \n            \n\n\n\n\n\n    \n...           1   
1\n            \n            \n\n\n\n\n\n    \n...           1   
1\n            \n            \n\n\n\n\n\n    \n...           1   
1\n            \n            \n\n\n\n\n\n    \n...           1   
1\n            \n            \n\n\n\n\n\n    \n...           2   
1\n            \n            \n\n\n\n\n\n    \n...           2   
1\n            \n            \n\n\n\n\n\n    \n...           1   
1\n            \n            \n\n\n\n\n\n    \n...           1   

                                                    gps_coordinates  accuracy  \
amount                                                                          
1                                                            186903       589   
1\n            \n            \n\n\n\n\n\n    \n...                1         1   
1\n            \n            \n\n\n\n\n\n    \n...                1         1   
1\n            \n            \n\n\n\n\n\n    \n...                1         1   
1\n            \n            \n\n\n\n\n\n    \n...                1         1   
1\n            \n            \n\n\n\n\n\n    \n...                2         1   
1\n            \n            \n\n\n\n\n\n    \n...                2         2   
1\n            \n            \n\n\n\n\n\n    \n...               12        10   
1\n            \n            \n\n\n\n\n\n    \n...                1         1   
1\n            \n            \n\n\n\n\n\n    \n...                1         1   

                                                    source    date  \
amount                                                               
1                                                       27  150394   
1\n            \n            \n\n\n\n\n\n    \n...       1       1   
1\n            \n            \n\n\n\n\n\n    \n...       1       1   
1\n            \n         

In [48]:
df_first_cleaning.groupby('date').nunique().head(10)

species_id  species_name  species_name_scientific  validation  \
date                                                                        
1930-09-28           1             1                        1           1   
1931-03-15           1             1                        1           1   
1931-10-24           1             1                        1           1   
1935-10-12           1             1                        1           1   
1945-10-16           1             1                        1           1   
1946-01-01           1             1                        1           1   
1946-08-04           1             1                        1           1   
1948-01-01           1             1                        1           1   
1948-01-07           1             1                        1           1   
1949-04-21           1             1                        1           1   

            gps_coordinates  accuracy  source  amount  life_stage  activity  \
date                                                                          
1930-09-28                1         1       0       1           1         1   
1931-03-15                1         1       0       1           1         1   
1931-10-24                1         1       0       1           1         1   
1935-10-12                1         1       0       1           1         1   
1945-10-16                1         1       1       1           1         1   
1946-01-01                1         1       1       1           1         1   
1946-08-04                2         1       1       2           1         1   
1948-01-01                1         1       1       1           1         1   
1948-01-07                1         1       1       1           1         1   
1949-04-21                1         1       1       1           1         1   

            location_id  location  observer_id  observer_name  \
date                                                            
1930-09-28            1         1            1              1   
1931-03-15            1         1            1              1   
1931-10-24            1         1            1              1   
1935-10-12            1         1            1              1   
1945-10-16            0         0            1              1   
1946-01-01            1         1            1              1   
1946-08-04            1         1            1              1   
1948-01-01            1         1            1              1   
1948-01-07            1         1            1              1   
1949-04-21            1         1            1              1   

            counting_method  method  
date                                 
1930-09-28                1       1  
1931-03-15                1       1  
1931-10-24                1       1  
1935-10-12                1       1  
1945-10-16                1       1  
1946-01-01                1       1  
1946-08-04                1       1  
1948-01-01                1       1  
1948-01-07                1       1  
1949-04-21                1       1

In [50]:
df_first_cleaning[df_first_cleaning['date'] == '1946-01-01'].head(2)

species_id species_name species_name_scientific validation  \
observation_id                                                               
78367649              70.0   Boomklever          Sitta europaea   Onbekend   

                gps_coordinates  accuracy source        date amount  \
observation_id                                                        
78367649        50.8486, 3.6375  onbekend   Site  1946-01-01      1   

               life_stage     activity  location_id  \
observation_id                                        
78367649         onbekend  ter plaatse     108154.0   

                                     location  observer_id observer_name  \
observation_id                                                             
78367649        Oudenaarde - Bos t'Ename (OV)      65638.0   Bos t'Ename   

               counting_method method  
observation_id                         
78367649              onbekend   geen

## Saven van de gecleande data

In [35]:
# Save the cleaned DataFrame to a new CSV file
df_first_cleaning.to_csv(os.path.join(folder_path_clean, f"observation_details_{species_id}_clean.csv"), index=False)